webapp/src/components/add_issue/add_issue.jsx

```diff
@@ -0,0 +1,332 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+    makeStyleFromTheme,
+    changeOpacity,
+} from 'mattermost-redux/utils/theme_utils';
+
+import TextareaAutosize from 'react-textarea-autosize';
+
+import FullScreenModal from '../modals/modals.jsx';
+import Button from '../../widget/buttons/button';
+import Chip from '../../widget/chip/chip';
+import AutocompleteSelector from '../user_selector/autocomplete_selector.tsx';
+import './add_issue.scss';
+import CompassIcon from '../icons/compassIcons';
+import {getProfilePicture} from '../../utils';
+
+const PostUtils = window.PostUtils;
+
+export default class AddIssue extends React.Component {
+    static propTypes = {
+        visible: PropTypes.bool.isRequired,
+        message: PropTypes.string.isRequired,
+        postID: PropTypes.string.isRequired,
+        assignee: PropTypes.object,
+        closeAddBox: PropTypes.func.isRequired,
+        submit: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        autocompleteUsers: PropTypes.func.isRequired,
+        openAssigneeModal: PropTypes.func.isRequired,
+        removeAssignee: PropTypes.func.isRequired,
+    };
+
+    shouldComponentUpdate() {
+        return true;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            message: '',
+            description: '',
+            sendTo: null,
+            attachToThread: false,
+            previewMarkdown: false,
+            assigneeModal: false,
+        };
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        if (props.visible && state.message == null) {
+            return {message: props.message};
+        }
+        if (!props.visible && (state.message != null || state.sendTo != null)) {
```

Comment:
Any reason why this won't work?

```suggestion
        if (!props.visible && (state.message || state.sendTo)) {
```